exporter/importables.go

```diff
@@ -1208,34 +1188,79 @@ var resourcesMap map[string]importable = map[string]importable{
 			name := r.ID[1:] + ext[language] // todo: replace non-alphanum+/ with _
 			content, _ := base64.StdEncoding.DecodeString(contentB64)
 			fileName, err := ic.createFileIn("notebooks", name, []byte(content))
-			splits := strings.Split(r.Name, "_")
-			notebookId := splits[len(splits)-1]
-			directorySplits := strings.Split(r.ID, "/")
-			directorySplits = directorySplits[:len(directorySplits)-1]
-			directoryPath := strings.Join(directorySplits, "/")
-
+			if err != nil {
+				return err
+			}
 			if ic.meAdmin {
 				ic.Emit(&resource{
 					Resource: "databricks_permissions",
-					ID:       fmt.Sprintf("/notebooks/%s", notebookId),
+					ID:       fmt.Sprintf("/notebooks/%d", r.Data.Get("object_id").(int)),
 					Name:     "notebook_" + ic.Importables["databricks_notebook"].Name(ic, r.Data),
 				})
 			}
 
 			// TODO: it's not completely correct condition - we need to make emit smarter - emit only if permissions are different from their parent's permission.
 			if ic.meAdmin {
+				directorySplits := strings.Split(r.ID, "/")
+				directorySplits = directorySplits[:len(directorySplits)-1]
+				directoryPath := strings.Join(directorySplits, "/")
+
 				ic.Emit(&resource{
 					Resource: "databricks_directory",
 					ID:       directoryPath,
 				})
 			}
 
+			log.Printf("Creating %s for %s", fileName, r)
+			r.Data.Set("source", fileName)
+			return r.Data.Set("language", "")
+		},
+		Depends: []reference{
+			{Path: "source", File: true},
+			{Path: "path", Resource: "databricks_user", Match: "home", MatchType: MatchPrefix},
+			{Path: "path", Resource: "databricks_service_principal", Match: "home", MatchType: MatchPrefix},
+		},
+	},
+	"databricks_workspace_file": {
+		Service: "notebooks",
+		Name:    workspaceObjectResouceName,
+		List:    createListWorkspaceObjectsFunc(workspace.File, "databricks_workspace_file", "workspace_file"),
+		Import: func(ic *importContext, r *resource) error {
```

Comment:
This is also almost exactly the same as for databricks_notebook. Is there a good reason for their implementations to diverge over time, or would it be better to have one function generator for importing both of these resources (like you do for list)?

Reply:
I was thinking about unifying, but there are a bit different implementations for writing the data & emitting permissions. I also have a request to export notebooks not only as source, but as `.ipynb` as well, so they will diverge more. But I need to think, maybe there could be a common part + function for actual data writing, etc.